server/channels/app/platform/service.go

```diff
@@ -339,6 +319,28 @@ func New(sc ServiceConfig, options ...Option) (*PlatformService, error) {
 	return ps, nil
 }
 
+// StartMetrics initializes thte metrics server if needed. It depends on the store
+// initialized and the license loaded, but PlatformService.New does not do the
+// latter, so the callers are responsible of making sure the license is loaded
+// before this function is called.
+func (ps *PlatformService) StartMetrics() error {
```

Comment:
Why can't we move this inside `(ps *PlatformService) Start()`? That way you don't have to expose any inner logic to app/server.go.